exec/cnex/global.c

```diff
@@ -604,9 +631,13 @@ void string__append(TExecutor *exec)
     Cell *addr = top(exec->stack)->address;
 
     addr->string->data = realloc(addr->string->data, addr->string->length + b->string->length);
+    if (addr->string->data == NULL) {
+        fatal_error("Could not allocate %d bytes for string append.",  addr->string->length + b->string->length);
+    }
     memcpy(&addr->string->data[addr->string->length], b->string->data, b->string->length);
     addr->string->length += b->string->length;
 
+    pop(exec->stack);
```

Comment:
Why pop this at the end? It would make more sense to pop it on line 631.

Reply:
If I did that, then I'd have to create a copy of the object.  Doing it this way, I saved that creation of the object,. and just used it as it existed on the stack, then pop the stack the end to destroy it, and free it.

Reply:
No, that's not true, because the cell that comes in on the stack is a *pointer* to the real object.

Reply:
If it does, that would be a serious bug. That's wouldn't be how pointers work in neon bytecode.